src/signin-signup/signup-form/index.tsx

```diff
@@ -0,0 +1,104 @@
+import { useRouter } from "next/router";
+import { useEffect, useState } from "react";
+import { useForm, SubmitHandler } from "react-hook-form";
+
+import {
+  isEmailUnique,
+  validateEmail,
+  validatePassword,
+} from "@/src/utils/validation";
+
+import handleSignUp from "./handleSignUp";
+import SignUpInput from "../input/signUpInput";
+import SignButton from "../sign-button";
+import S from "../signin-form/SignForm.module.scss";
+
+export interface SignUpFormValues {
+  email: string;
+  password: string;
+  passwordCheck: string;
+}
+
+const SignupForm = () => {
+  const router = useRouter();
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+    watch,
+  } = useForm<SignUpFormValues>({ mode: "onBlur" });
+
+  const [emailError, setEmailError] = useState(false);
+  const [passwordError, setPasswordError] = useState(false);
+  const [passwordCheckError, setPasswordCheckError] = useState(false);
+
+  useEffect(() => {
+    errors.email ? setEmailError(true) : setEmailError(false);
+    errors.password ? setPasswordError(true) : setPasswordError(false);
+    errors.passwordCheck
+      ? setPasswordCheckError(true)
+      : setPasswordCheckError(false);
+  }, [errors.email, errors.password, errors.passwordCheck]);
+
+  // 회원가입
+  const submitSignUp: SubmitHandler<SignUpFormValues> = (data) => {
+    const email = data.email;
+    const password = data.password;
+
+    if (
+      emailError === false &&
+      passwordError === false &&
+      passwordCheckError === false
+    ) {
+      handleSignUp({ email, password }).then((res) => {
+        if (res === true) {
+          router.push("/folder");
+        }
+      });
+    }
```

Comment:
early return하면 더 깔끔할 것 같습니다.
```suggestion
    const isAllValid = !emailError && !passwordError && !passwordCheckError;
    
    if(!isAllValid) return;
    
    const email = data.email;
    const password = data.password;
    
    handleSignUp({ email, password }).then((res) => {
      if (res) {
        router.push("/folder");
      }
    });
  }
```